### PR TITLE
added more configuration options to gitit to allow github authentication

### PR DESCRIPTION
**Pull request by statusfailed**
These are extra options available in the gitit config file- see the gitit [README](https://github.com/jgm/gitit) for details

### CODE REVIEW

File: nixos/modules/services/misc/gitit.nix [hunk 1]
```diff
@@ -587,6 +623,14 @@ video/x-ms-wmx  wmx
     pdf-export: ${toYesNo cfg.pdfExport}
     pandoc-user-data: ${toString cfg.pandocUserData}
     xss-sanitize: ${toYesNo cfg.xssSanitize}
+
+    [Github]
+    oauthclientid: ${cfg.oauthClientId}
```
Comment (peti) [inline nixos/modules/services/misc/gitit.nix]: What happens when these variables are set to `null`? Isn't the result an evaluation error?

**Comment by jagajaga** [comment]
Please rename commit's name following our guide in `CONTRIBUTING.md`

File: nixos/modules/services/misc/gitit.nix (hunk 1 above)
Comment (statusfailed) [inline nixos/modules/services/misc/gitit.nix]: Yes, you're right- thanks!

**Comment by statusfailed** [comment]
Is it just the commits that need to be renamed, or the pull request too?

**Comment by jagajaga** [comment]
You did everything nicely :)
Thank you.

**Pull request merged**